### ISSUE TITLE
[4.0.0-beta.11] Dynamically updated content needs to be wrapped in ngZone.run() in order to update

### ISSUE THREAD

**Issue by jordanpurinton** [ionic-team/ionic#15765]
# Bug Report

**Ionic Info**

```
Ionic:

   ionic (Ionic CLI)          : 4.1.2 (/usr/local/lib/node_modules/ionic)
   Ionic Framework            : @ionic/angular 4.0.0-beta.11
   @angular-devkit/core       : 0.8.3
   @angular-devkit/schematics : 0.8.3
   @angular/cli               : 6.2.3
   @ionic/ng-toolkit          : 1.0.8
   @ionic/schematics-angular  : 1.0.6

Cordova:

   cordova (Cordova CLI) : 8.0.0
   Cordova Platforms     : android 7.1.1
   Cordova Plugins       : cordova-plugin-ionic-keyboard 2.1.2, cordova-plugin-ionic-webview 2.1.4, (and 10 other plugins)

System:

   Android SDK Tools : 26.1.1 (/path/Android/Sdk)
   NodeJS            : v8.10.0 (/usr/bin/node)
   npm               : 6.4.1
   OS                : Linux 4.15
```

**Describe the Bug**
Upon upgrading to the newest version of `@ionic/angular`, several instances of my app where I am dynamically updating content via API now require to be wrapped in an `ngZone.run()` call in order to work. Additionally, several instances don't even seem to work when wrapped in this call.

**Related Code**
<a href="https://gist.github.com/jordanpurinton/6e88f43c200db4c0b78276fcb8eb4a15">Example of list update requiring an `ngZone.run()` call</a>

**Expected Behavior**
For the list to behave as it previously did and not require to be wrapped in `ngZone.run()` when dynamically updating content.

**Comment by mhartington** [comment]
Hi there! Can you provide a more complete example of this error and push it to github?

**Comment by jordanpurinton** [comment]
@mhartington Hey! Working on it atm.

**Comment by jordanpurinton** [comment]
Let me know if <a href="https://github.com/jordanpurinton/ngzone-ionic-beta12">this</a> is working for ya @mhartington. If you attempt to filter the list, it won't update unless you tap the filter button again.

**Comment by quynh-ng** [comment]
I got same issue, some times the view was updated, some times not (to see any updated, need to click/tap any button/control on the view).

Currently workaround is manual call *detectChanges()* of ***ChangeDetectorRef***.

**Comment by jordanpurinton** [comment]
@quynh-ng Yeah, I've even had success using beta.7, but anything after that seems to throw a fit.

**Comment by mhartington** [comment]
So @jordanpurinton, I'm not seeing the error with the latest beta. I clone your repo, ran npm install and things worked as expected

Here I tested your example with and without zone.run and all works as expect. 


![zone-check](https://user-images.githubusercontent.com/2835826/46168628-040a3d00-c267-11e8-8c54-90119ca15a79.gif)

**Comment by RHinderiks** [comment]
We too have to apply applicationRef.tick() to see changes when for example changing the language or picking a date form ion-datetime.

**Comment by jordanpurinton** [comment]
Yeah @RHinderiks, this also applies to me on some inputs on versions greater than beta.7.

**Comment by quynh-ng** [comment]
In my case, this bug only occurred when the page (with dynamic content) is navigated by a handler function of an action sheet item (means navigated by code via navigateForward method of NavController).

Details in step-by-step:
- Step 1: User clicks/taps a button (on the home screen) to tell Ionic shows the action-sheet (with handler is a function to navigate to page of items => page with dynamic contents)
- Step 2: Ionic's ActionSheetController creates and shows the action-sheet
- Step 3: When user clicks/taps into an item of the action-sheet, the navigating process occurred by call the **navigateForward** method of NavController. In this step, the execution context (NgZone) for running the handler function is action-sheet's context - means is **NOT the NgZone** of current page. That why Angular shows the warning: *Navigation triggered outside Angular zone, did you forget to call 'ngZone.run()'?*.
- Step 4: the page of items (page with dynamic content) is shown. At the first-time everything work as expected, but Angular can't detect any changes after, and the view is only updated when user clicks/taps into any controls/inputs/... or manually call the method **detectChanges()** of ChangeDetectorRef.

Now let modify some line-of-code of handler function for navigating at step 3: wrap the call of the **navigateForward** method (of NavController) into NgZone.run() to make sure the execution context is current NgZone of the page, Angular doesn't show the warning and detects all changes (at the step 4). That why @jordanpurinton (and I) need a workaround => Dynamically updated content needs to be wrapped in ngZone.run() in order to update.

Here my action-sheet:
```typescript
this.actionsheetController.create({
	buttons: [
		{
			text: 'Indirect navigate - by NgZone.run()',
			handler: () => this.zone.run(() => {
				this.navController.navigateForward('/items', true);
			})
		},
		{
			text: 'Direct Navigate - without NgZone.run()',
			handler: () => {
				this.navController.navigateForward('/items', true);
			}
		}
	]
}).then(async actionsheet => await actionsheet.present());
```

Please clone this repo and see what happen with two context/NgZone => https://github.com/quynh-ng/ionic-bugs-changes-not-detected
